lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -1038,6 +1038,20 @@ namespace Js
         return GetExternalDisplayName(this);
     }
 
+    JavascriptString* ParseableFunctionInfo::GetExternalDisplayNameObject(ScriptContext* scriptContext) const
+    {
+        const char16* name = GetExternalDisplayName();
+
+        if (!GetDisplayNameIsRecyclerAllocated() && !IsConstantFunctionName(name))
+        {
+            return Js::JavascriptString::NewCopySz(name, scriptContext);
```

Comment:
This would mess with the const-ness, but is it worth updating the `this->m_displayName` which is returned from `GetExternalDisplayName` to use this recycler allocated string?

Reply:
I thought about that, but since the string can shortcut to a per-ScriptContext empty or single-character string it seems safer to just create it each time.